src/neo/Trie/MPT/MPTProofStore.cs

```diff
@@ -0,0 +1,39 @@
+using Neo.Cryptography;
+using Neo.IO;
+using Neo.Persistence;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Neo.Trie.MPT
+{
+    public class MPTProofStore : IReadOnlyStore
+    {
+        private Dictionary<byte[], byte[]> store = new Dictionary<byte[], byte[]>(ByteArrayEqualityComparer.Default);
+
+        public MPTProofStore(HashSet<byte[]> proof)
+        {
+            foreach (byte[] data in proof)
+            {
+                store.Add(Crypto.Hash256(data), data);
+            }
+        }
+
+        public byte[] TryGet(byte prefix, byte[] hash)
+        {
+            var result = store.TryGetValue(hash, out byte[] value);
+            return result ? value : null;
+        }
+
+        public IEnumerable<(byte[] Key, byte[] Value)> Find(byte table, byte[] prefix)
+        {
+            foreach (var pair in store)
+            {
+                if (prefix is null || pair.Key.CommonPrefix(prefix).Length == prefix.Length)
```

Comment:
You can use `StartsWith()`. Here is an example:

https://github.com/neo-project/neo/blob/8b558d6bc92123ec78c4a1aed3bceb4c4f892085/src/neo/IO/Caching/DataCache.cs#L148-L157